ftva_lab_data/views_utils.py

```diff
@@ -214,3 +238,52 @@ def get_items_per_page_options() -> list[int]:
     """
 
     return [10, 20, 50, 100]
+
+
+def format_data_for_export(data_dicts: list[dict[str, Any]]) -> pd.DataFrame:
+    """Formats a list of dictionaries of SheetImport data for export to Excel.
+
+    :param data_dicts: A list of dictionaries, each representing a row of data.
+    :return: A pandas DataFrame with the formatted data.
+    """
+
+    # Add and remove fields to match the expected output format:
+    # Add a column for Status (ManyToMany relationship with Status model),
+    # replace the Assigned User ID with assigned_user_full_name property, and
+    # remove the '_state' field added by Django.
+    for data_dict in data_dicts:
+        current_item = SheetImport.objects.get(id=data_dict["id"])
+        # Add status display values as a concatenated string
+        statuses = current_item.status.values_list("status", flat=True)
+        data_dict["status"] = ", ".join(statuses) if statuses else ""
+        # Replace the assigned_user_id with the full name
+        assigned_user_id = data_dict.pop("assigned_user_id")
+        if assigned_user_id is not None:
+            assigned_user_full_name = current_item.assigned_user_full_name
+            data_dict["assigned_user"] = assigned_user_full_name
+        else:
+            data_dict["assigned_user"] = ""
+        # Remove the '_state' field added by Django
+        data_dict.pop("_state", None)
+
+    # Convert rows to DataFrame for exporting
+    df = pd.DataFrame(data_dicts)
+
+    # Reorder carrier_a_location and carrier_b_location to be next to carrier_a and carrier_b
```

Comment:
This code is not needed, if you change the order of the fields in the model.  I checked by disabling this code and moving the location fields, like this:
```
    carrier_a = models.CharField(max_length=100, blank=True)
    carrier_a_location = models.CharField(max_length=25, blank=True)
    carrier_b = models.CharField(max_length=100, blank=True)
    carrier_b_location = models.CharField(max_length=25, blank=True)
```
Removing the carrier reordering code also will resolve several "Operator not supported" type-related errors pylance is showing with these lines of code.